x-pack/filebeat/input/cel/config.go

```diff
@@ -122,6 +114,18 @@ func (c config) Validate() error {
 	return nil
 }
 
+// checkUnsupportedParams checks if unsupported/deprecated/discouraged paramaters are set and logs a warning
+func (c config) checkUnsupportedParams(logger *logp.Logger) {
+	if c.RecordCoverage {
+		logp.L().Named("cel").Warn("execution coverage enabled: " +
+			"see documentation for details: https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-cel.html#cel-record-coverage")
+	}
+	if c.Redact == nil {
+		logp.L().Named("cel").Warn("missing recommended 'redact' configuration: " +
+			"see documentation for details: https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-cel.html#cel-state-redact")
+	}
+}
+
```

Comment:
Please move this to under its caller in input_manager.go.

Also note that logger is not used here (and prefer s/logger/log/g in this package).